sway/border.c

```diff
@@ -8,21 +8,16 @@
 #include "config.h"
 #include "client/pango.h"
 
+#include <endian.h>
+
 void cairo_set_source_u32(cairo_t *cairo, uint32_t color) {
-	int endian = 1;
-	if (*(char *)&endian == 1) { // little endian
-		cairo_set_source_rgba(cairo,
-				(color >> (1*8) & 0xFF) / 255.0,
-				(color >> (2*8) & 0xFF) / 255.0,
-				(color >> (3*8) & 0xFF) / 255.0,
-				(color >> (0*8) & 0xFF) / 255.0);
-	} else { // big endian
-		cairo_set_source_rgba(cairo,
-				(color >> (0*8) & 0xFF) / 255.0,
-				(color >> (3*8) & 0xFF) / 255.0,
-				(color >> (2*8) & 0xFF) / 255.0,
-				(color >> (1*8) & 0xFF) / 255.0);
-	}
+    color = htole32(color);
+
+    cairo_set_source_rgba(cairo,
+            (color >> (1*8) & 0xFF) / 255.0,
+            (color >> (2*8) & 0xFF) / 255.0,
+            (color >> (3*8) & 0xFF) / 255.0,
+            (color >> (0*8) & 0xFF) / 255.0);
```

Comment:
Scratch that, I was wrong all along, and the cairo_set_source_rgba is also wrong..